### PR TITLE
pounder: support ext ref, refactor gpio logic

### PR DESCRIPTION
**Pull request by jordens**

### CODE REVIEW

File: src/hardware/pounder/mod.rs [hunk 1]
```diff
@@ -360,36 +364,44 @@ impl PounderDevices {
...
-        let pin = channel as u8;
-        self.mcp23017
-            .write_gpio(mcp23017::Port::GPIOB, 0x2f & !(1 << pin))
-            .map_err(|_| Error::I2c)?;
-        // Duration of one I2C transaction is sufficiently long.
-        self.mcp23017
-            .write_gpio(mcp23017::Port::GPIOB, 0x2f)
-            .map_err(|_| Error::I2c)?;
-        Ok(())
+        let pin = GpioPin::try_from(8 + channel as u8).unwrap();
```
Comment (ryan-summers) [inline src/hardware/pounder/mod.rs:401]:
```suggestion
        let pin = GpioPin::try_from(GpioPin::AttLe0 as u8 + channel as u8).unwrap();
```

**Review by ryan-summers — COMMENTED** [review]
One nit on using the constant 8 instead of a pin index. Can honestly go either way on it though.

File: src/hardware/pounder/mod.rs (hunk 1 above)
Comment (jordens) [reply src/hardware/pounder/mod.rs]: Nice

**Comment by jordens** [comment]
bors r=ryan-summers

**Comment by jordens** [comment]
bors r=ryan-summers

**Pull request merged**